cub/device/dispatch/dispatch_batch_memcpy.cuh

```diff
@@ -316,6 +326,7 @@ template <typename InputBufferIt,
           typename BufferSizeIteratorT,
           typename BufferOffsetT,
           typename BlockOffsetT,
+          bool memcpy             = true,
```

Comment:
When the copy is called for a type whose assignment operator has been overloaded, won't this create a problem? Would I need to check both that it is a pointer and it is trivially copyable?

Reply:
Also, even if we infer from the types, the buffer sizes will be incorrect for trivially copyable types if a user called the Copy instead of Memcpy. How can we know whether Copy or Memcpy was called?

Reply:
>Also, even if we infer from the types, the buffer sizes will be incorrect for trivially copyable types if a user called the Copy instead of Memcpy.

Thanks for bringing this up. You are right. We should maintain the `memcpy` template parameter to be able to identify whether the user has called the `DeviceCopy` or `DeviceMemcpy` interface and, hence, to correctly infer whether `size` would refer to `number_of_bytes` or `number_of_items`.

Could you put the `memcpy` template parameter last, though. That's to make sure that, if previously somebody was using the `Dispatch` class by providing the explicit template arguments, the template parameter order is unchanged and just extended.